doc/specs/#605 - Search/spec.md

```diff
@@ -19,41 +19,56 @@ One of the superior features of iTerm2 is it's content search. The search comes
 2. If we put the search box under TerminalApp, then the search can only happen on the current focused Terminal. 
 3. If the community does not like the current design, we can lift SearchBox to a higher level. 
 
-### Search process
-1. The user press Ctrl+F (or user's custom key binding) to open the search box. Focus will move to the TextBox. 
+### Search process implementation
+1. Once the user press <kbd>ctrl+shift+f</kbd> (or user's custom key binding), a new `SearchBoxControl` object will be created and attached as a child of `TermControl`, Focus will move to the TextBox within the `SearchBoxControl`. 
 2. Search is performed on a XAML TextBox. Once the user presses Enter or click up/down arrow button, we start to search from the last line of the current viewport or the current selection, and try to find the exact text in the text buffer. The nearest searched one will be selected. Then the search start point will be set to the selected text. The next search will start before or after the previous searched text.
-3. The user can choose to search up or down by choosing up arrow or down arrow buttons. The chosen button will be styled to indicate it is selected. If the user does not click the arrows buttons, the default direction is up. 
-4. The user can choose to do case sensitive or insensitive match by checking a check box. The default is case insensitive. 
-5. If the search box is focused, the user can navigte all the elements on the search box using tab. When selected, press Enter equals to click. 
-6. If the user click the "X" button or press <kbd>Esc</kbd>, the search stopped and the search box disappears and focus will move back to Terminal. In phrase one we do not store any state. 
-7. Once the search box is closed (exiting search mode), the selection will still be there. This coincides with the current VS Code and cmd experience. To get rid of the selection, the user can just click other area of the window.
+3. We re-use the Search module in conhost. It performs the search in a brute-force approach. Starting from every position in the text buffer, the search algorithm compares the span of the searched string with buffer characters, and if the current buffer text matches the whole string, it will return store the position of the text in the buffer and return. The stored position information will be used for selection. 
+3. The user can choose to search up or down. Search module realizes this, we just need to set a boolean flag. Default is search up. 
+4. The user can choose to do case sensitive or insensitive match. This also realized by Search module by setting a boolean flag. Default is search case-insensitively. 
+5. Tab navigation is realized by XAML. We just need to set TabNavigation="Cycle" in `SearchBoxControl`. 
+6. If the user click the "X" button or press <kbd>Esc</kbd>, the search box will disappear and the object will be destructed and detached from the `TermControl` XAML tree. In phrase one we do not store any state. 

```

Comment:
This line looks like it was moved down to L67, so it can probably be removed here

Reply:
I think this line focuses on the algorithmic perspective while line 67 focuses on the behavior that the user could be see.